### PR TITLE
Docs: sync breaking changes from release notes

### PR DESCRIPTION
**Pull request by andreidan**
Add those breaking changes present into the release notes but missing
from the migration notes.

### Preview
https://elasticsearch_63479.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/7.10/breaking-changes-7.10.html

### CODE REVIEW

File: docs/reference/migration/migrate_7_10.asciidoc [hunk 1]
```diff
@@ -18,54 +18,131 @@ See also <<release-highlights>> and <<es-release-notes>>.
...
-and was never released. This API is superseded by the <<repositories-metering-apis,Repositories Metering APIs>>
-added in 7.10.0 which should be used instead. The Repository Stats API is
-deprecated starting 7.10.0 and will be removed in 8.0.0.
-
+For future feature development, we added a `Supplier<SearchLookup>` argument to
+the existing `MappedFieldType#fielddataBuilder` method.
+ 
 *Impact* +
-Use the <<repositories-metering-apis,Repositories Metering APIs>>.
+No action needed.
```
Comment (jrodewig) [inline docs/reference/migration/migrate_7_10.asciidoc:68]: @javanna  Can you confirm this is the case? If Java users need to make a change to accommodate the new argument, I'd like to give them next steps here.

File: docs/reference/migration/migrate_7_10.asciidoc [hunk 2]
```diff
@@ -18,54 +18,131 @@ See also <<release-highlights>> and <<es-release-notes>>.
...
+==== Java changes
 
-//end::notable-breaking-changes[]
+[discrete]
+==== Machine Learning: `allow_no_jobs` and `allow_no_datafeeds` deprecated in favor of `allow_no_match`
+`allow_no_jobs` and `allow_no_datafeeds` API parameters are deprecated in favor
+of `allow_no_match`. The old parameters are still accepted by the APIs but the
+deprecation warning is emitted if the old parameter name is used in the request
+body or as a request param.
+High-level REST client classes will now send the new `allow_no_match` parameter.
```
Comment (jrodewig) [inline docs/reference/migration/migrate_7_10.asciidoc:56]: @lcawl Could you take a look at the content of this and reformat?

**Review by jrodewig — APPROVED** [review]
LGTM overall. I updated several of the sections to use the new format. I also reordered them.

I left one change for @lcawl to handle.

I also left a question for @javanna  re: the Java changes for runtime fields.

Thanks @andreidan!

File: docs/reference/migration/migrate_7_10.asciidoc (hunk 1 above)
Comment (javanna) [reply docs/reference/migration/migrate_7_10.asciidoc:68]: the change impacts plugin developers that plug in a custom field type, meaning that their impl most likely defines the fielddataBuilder method which has now a different signature. They will have to adapt the signature of the method they override, that's it. This is one of those breaking changes thats easily solvable thanks to the java compiler

Comment (jrodewig) [reply docs/reference/migration/migrate_7_10.asciidoc]: Thanks @javanna. I added some text to address. Let me know if I missed anything or left out an important detail.

File: docs/reference/migration/migrate_7_10.asciidoc [hunk 3]
```diff
@@ -18,54 +18,133 @@ See also <<release-highlights>> and <<es-release-notes>>.
...
-added in 7.10.0 which should be used instead. The Repository Stats API is
-deprecated starting 7.10.0 and will be removed in 8.0.0.
-
+To support future feature development, the existing
+`MappedFieldType#fielddataBuilder` method now accepts a new
+`Supplier<SearchLookup>` argument.
+ 
 *Impact* +
-Use the <<repositories-metering-apis,Repositories Metering APIs>>.
+If developing or maintaining an {es} plugin, update your implementation of the
```
Comment (javanna) [inline docs/reference/migration/migrate_7_10.asciidoc:68]: maybe we can be more precise and say "if you implement a mapper plugin" as not all plugins need to implement that method. But maybe it's too much, up to you :)

Comment (jrodewig) [reply docs/reference/migration/migrate_7_10.asciidoc]: Thanks for the feedback! I updated to `If you develop or maintain a mapper plugin,...`

Comment (javanna) [reply docs/reference/migration/migrate_7_10.asciidoc]: thank you!

File: docs/reference/migration/migrate_7_10.asciidoc [hunk 4]
```diff
@@ -18,54 +18,171 @@ See also <<release-highlights>> and <<es-release-notes>>.
...
+* `POST _cluster/reroute`
+* `GET {index}/_stats`
+* `GET {index}/_segments`
+* `GET {index}/_shard_stores`
+* `GET _cat/[indices,aliases,health,recovery,shards,segments]`
+
+*Impact* +
+To avoid deprecation warnings, do not use unsupported REST APIs to access system
+indices.
+====
```
Comment (jrodewig) [inline docs/reference/migration/migrate_7_10.asciidoc:81]: CC @gwbrown

**Comment by andreidan** [comment]
@lcawl could you take a look at the machine learning content please?

File: docs/reference/migration/migrate_7_10.asciidoc [hunk 5]
```diff
@@ -18,54 +18,171 @@ See also <<release-highlights>> and <<es-release-notes>>.
...
+
 
 [discrete]
-==== *.tcp.keep_idle and *.tcp.keep_interval now bounded
-The settings `{network,transport,http}.tcp.keep_idle` and
-`{network,transport,http}.tcp.keep_interval` are now limited to a maximum
-value of 300, which amounts to 5 minutes. This helps ensure that, if
-explicitly configured, only reasonable values are set.
+[[breaking_710_java_changes]]
+==== Java changes
```
Comment (lcawl) [inline docs/reference/migration/migrate_7_10.asciidoc:86]: > Java changes

This section seems to be blank.  Should it be removed?

File: docs/reference/migration/migrate_7_10.asciidoc [hunk 6]
```diff
@@ -18,54 +18,182 @@ See also <<release-highlights>> and <<es-release-notes>>.
...
+
+*Impact* +
+To avoid errors, specify the `name` property when creating or granting API keys.
+====
 
 [discrete]
-[[breaking_710_networking_changes]]
-=== Networking Changes
+[[breaking_710_indices_changes]]
+==== Indices changes
```
Comment (lcawl) [inline docs/reference/migration/migrate_7_10.asciidoc:48]: I think this should be a peer section to the Authentication section, not a child.
```suggestion
=== Indices changes
```

File: docs/reference/migration/migrate_7_10.asciidoc [hunk 7]
```diff
@@ -18,54 +18,182 @@ See also <<release-highlights>> and <<es-release-notes>>.
...
+
 
 [discrete]
-==== *.tcp.keep_idle and *.tcp.keep_interval now bounded
-The settings `{network,transport,http}.tcp.keep_idle` and
-`{network,transport,http}.tcp.keep_interval` are now limited to a maximum
-value of 300, which amounts to 5 minutes. This helps ensure that, if
-explicitly configured, only reasonable values are set.
+[[breaking_710_java_changes]]
+==== Java changes
```
Comment (lcawl) [inline docs/reference/migration/migrate_7_10.asciidoc:86]: Likewise, I think this should be a peer section to the others.
```suggestion
=== Java changes
```

**Review by lcawl — APPROVED** [review]
LGTM, thanks!

**Pull request merged**